core/controllers/editor_test.py

```diff
@@ -412,6 +412,75 @@ def _submit_answer(
             self.logout()
 
 
+class ExplorationEditorLogoutTest(BaseEditorControllerTest):
+    """Test handler for logout from exploration editor page."""
+
+    def test_logout_from_invalid_url(self):
+        """Logout from invalid exploration id should result
+        in 404 error.
+        """
+
+        invalid_current_page = '%s/%s' % (
+            feconf.EDITOR_URL_PREFIX, 'invalid_eid')
+        invalid_logout_url = (
+            '/exploration_editor_logout?return_url=%s' % invalid_current_page)
+
+        self.login(self.OWNER_EMAIL)
+        response = self.testapp.get(invalid_logout_url, expect_errors=True)
+        self.assertEqual(response.status_int, 404)
```

Comment:
Why does this result in 404 and not a redirect to the library page?

Also, there are two different logout cases: an empty return_url, and a nonexistent exp id. Test both cases separately.

Reply:
i have changed it to redirect to library.